### PR TITLE
Source RKI-Covid - add history data, incidence rate, cases, death, hospitalization streams

### PR DESCRIPTION
**Pull request by Zawar92**
## What
n the connector [source-rki-covid](https://github.com/airbytehq/airbyte/tree/master/airbyte-integrations/connectors/source-rki-covid) we want to add streams for the states that include history data, incidence rate, cases, deaths, hospitalization.

**Related Issue**: https://github.com/airbytehq/airbyte/issues/16892

## How
Adding new streams for states.

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [X] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [X] Secrets in the connector's spec are annotated with `airbyte_secret`
- [X] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [X] Code reviews completed
- [ ] Documentation updated
    - [X] Connector's `README.md`
    - [X] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

unit_test

</details>

<details><summary><strong>Integration</strong></summary>

Integration-Test

</details>

<details><summary><strong>Acceptance</strong></summary>

Acceptance test

</details>

### CODE REVIEW

**Comment by Zawar92** [comment]
The `conflict's`  seen above occurred when **syncing with new pull** from `Airbyte`. Hope these not cause any `problem`.

**Comment by Zawar92** [comment]
Build.
build

**Comment by sajarin** [comment]
Thanks for the PR @Zawar92, I cleaned it up for you. This PR is part of Airbyte's Community Maintainer program and will be reviewed and merged by a member of our community. We ask for your patience as you wait for someone to be assigned to this PR! Thanks again :)

**Comment by sajarin** [comment]
assigned to @cptjacky

**Review by cptjacky — COMMENTED** [review]
Hello @Zawar92 and thank you for your contribution!

I left a few comments below to address before we can merge this!

Could you also update the `docs/integrations/sources/rki-covid.md` file with the new added streams? (both in the documentation part & in the changelog)

If you need any help understanding some of my comments, feel free to reach out to me!

File: airbyte-integrations/connectors/source-rki-covid/source_rki_covid/source.py
Comment (Zawar92) [reply airbyte-integrations/connectors/source-rki-covid/source_rki_covid/source.py:440]: **Understood**

File: airbyte-integrations/connectors/source-rki-covid/source_rki_covid/source.py
Comment (Zawar92) [reply airbyte-integrations/connectors/source-rki-covid/source_rki_covid/source.py:435]: Understood

File: airbyte-integrations/connectors/source-rki-covid/unit_tests/test_incremental_stateshistorycases.py
Comment (Zawar92) [reply airbyte-integrations/connectors/source-rki-covid/unit_tests/test_incremental_stateshistorycases.py:13]: **Understood**

File: airbyte-integrations/connectors/source-rki-covid/integration_tests/configured_catalog.json
Comment (Zawar92) [reply airbyte-integrations/connectors/source-rki-covid/integration_tests/configured_catalog.json:614]: **Understood**

File: airbyte-integrations/connectors/source-rki-covid/integration_tests/sample_state.json
Comment (Zawar92) [reply airbyte-integrations/connectors/source-rki-covid/integration_tests/sample_state.json:21]: Should I `remove` from Line 21-41? That what I understood.

File: airbyte-integrations/connectors/source-rki-covid/source_rki_covid/schemas/states_history_cases.json
Comment (Zawar92) [reply airbyte-integrations/connectors/source-rki-covid/source_rki_covid/schemas/states_history_cases.json:5]: I didn't get this. Can you explain?

File: airbyte-integrations/connectors/source-rki-covid/source_rki_covid/schemas/germany_states_age_groups.json
Comment (Zawar92) [reply airbyte-integrations/connectors/source-rki-covid/source_rki_covid/schemas/germany_states_age_groups.json:8]: I didn't get this. Can you explain?

File: airbyte-integrations/connectors/source-rki-covid/source_rki_covid/schemas/states_history_cases.json
Comment (Zawar92) [reply airbyte-integrations/connectors/source-rki-covid/source_rki_covid/schemas/states_history_cases.json:5]: I would be getting a `schema` error in acceptance test.

**Comment by Zawar92** [comment]
@cptjacky,

I need some clarification in some comments. Can you give me an answer?

**Comment by cptjacky** [comment]
> @cptjacky,
> 
> I need some clarification in some comments. Can you give me an answer?

Hey @Zawar92, I am sorry I did not answer earlier. I will clarify the comments in depth tomorrow afternoon.

File: airbyte-integrations/connectors/source-rki-covid/integration_tests/sample_state.json
Comment (cptjacky) [reply airbyte-integrations/connectors/source-rki-covid/integration_tests/sample_state.json:21]: Yes, you should remove them!

File: airbyte-integrations/connectors/source-rki-covid/source_rki_covid/schemas/germany_states_age_groups.json
Comment (cptjacky) [reply airbyte-integrations/connectors/source-rki-covid/source_rki_covid/schemas/germany_states_age_groups.json:8]: Currently, the data looks like:

```json
{
   "state":"xx",
   "data":{
      "AXX-AYY":{
         "..."
      }
   }
}
```

I was suggesting to make it look like
```json
{
   "state":"xx",
   "AXX-AYY":{
      "..."
   }
}
```

(removing one level of nesting)

File: airbyte-integrations/connectors/source-rki-covid/source_rki_covid/schemas/states_history_cases.json
Comment (cptjacky) [reply airbyte-integrations/connectors/source-rki-covid/source_rki_covid/schemas/states_history_cases.json:5]: Your code is choosing these keys, in the `source.py`.

You should:
- Rename the fields here like I mentioned above. (for all `states` streams)
- Change the logic in `source.py` to match your new schema (for all `states` streams)
- Adapt acceptance tests if needed.

File: airbyte-integrations/connectors/source-rki-covid/source_rki_covid/source.py
Comment (cptjacky) [inline airbyte-integrations/connectors/source-rki-covid/source_rki_covid/source.py:456]: To follow up on https://github.com/airbytehq/airbyte/pull/17607#discussion_r1006064120, this here would look like:

```suggestion
                    x.update({"name": value.get("name"), "abbreviation": key})
```

or 
```suggestion
                    x.update({"state_name": value.get("name"), "state_abbreviation": key})
```

**Comment by Zawar92** [comment]
Build

**Comment by cptjacky** [comment]
Hello @Zawar92! Thank you for making the changes, I will be taking a closer look at them tomorrow!

**Comment by marcosmarxm** [comment]
Hello :wave:, first thank you for this amazing contribution. 

We really appreciate the effort you've made to improve the project.
We ask you patience for the code review. Last month our team was focused on Hacktoberfest event and that probably left some PR without the proper feedback. And this week, due to the Thanksgiving US Holiday, most our team is out of office with their families. Another important piece of information why code won't be merge this week is: as a safety measure the core team has decided to freeze merging code to main branch to keep the release stable. Next week we'll return to you with the proper code review and update the status of your contribution.

If you have any questions feel free to send me a message in Slack!
Thanks!

File: airbyte-integrations/connectors/source-rki-covid/source_rki_covid/source.py
Comment (cptjacky) [inline airbyte-integrations/connectors/source-rki-covid/source_rki_covid/source.py:98]: Small suggestion that should be equivalent

```suggestion
            for key, record in response.json().get("data").items():
                record["abbreviation"] = key
                yield record
```

**Review by cptjacky — APPROVED** [review]
Hello @Zawar92 ! Sorry for the daly, and thank you for taking my comments into account! 

The changes look good! I just made one last optional suggestion.

**Comment by cptjacky** [comment]
There's just one thing I am not sure about: @sajarin, are the formatting changes to other connectors an issue?

If yes, we'll probably need to remove those from this PR, otherwise, I'm approving this PR!

**Comment by sajarin** [comment]
Hey @cptjacky, 

For the sake of simplicity, we should remove the formatting changes to other connectors. It makes it difficult to interpret what this PR is changing from looking at the commit history. It also increases the chance for this PR to break other connectors, so let's keep it just related to the RKI-Covid source.

File: airbyte-integrations/connectors/source-rki-covid/source_rki_covid/source.py
Comment (Zawar92) [reply airbyte-integrations/connectors/source-rki-covid/source_rki_covid/source.py:98]: Thanks @cptjacky, I will look into those.

Comment (Zawar92) [reply airbyte-integrations/connectors/source-rki-covid/source_rki_covid/source.py:98]: Hi @cptjacky. Can I fix this last `optional change` `later` when there is time. As the current change is very important to us and our clients and we require the `state data` streams that I have added.

Comment (cptjacky) [reply airbyte-integrations/connectors/source-rki-covid/source_rki_covid/source.py:98]: Yes we can leave this like this!

However, as mentioned in Saj's comment, please remove all the formatting changes affecting other connectors from your PR before we can merge it!

**Comment by Zawar92** [comment]
Hi @cptjacky. Can I fix this last optional `change later when there is time`. As the current change is very `important` to us and our `clients` and we require the `state data streams `that I have added.

I hope this `change is not a compulsory` one and I can look at it after this merge.

**Comment by cptjacky** [comment]
> Hi @cptjacky. Can I fix this last optional `change later when there is time`. As the current change is very `important` to us and our `clients` and we require the `state data streams `that I have added.
> 
> I hope this `change is not a compulsory` one and I can look at it after this merge.

Hey @Zawar92,

See my message in another thread:

> Yes we can leave this like this!
> 
> However, as mentioned in Saj's comment, please remove all the formatting changes affecting other connectors from your PR before we can merge it!

You need to remove the unnecessary formating changes to other connectors before we merge this.

**Comment by Zawar92** [comment]
Hi @cptjacky,

I took a rebase from the main branch to remove the unnecessary formating changes.

Does this sound fine.

**Comment by sajarin** [comment]
@Zawar92 I cleaned up your branch, could you double check to make sure I haven't missed any of the changes you've previously committed to the branch?

**Comment by Zawar92** [comment]
> @Zawar92 I cleaned up your branch, could you double check to make sure I haven't missed any of the changes you've previously committed to the branch?

Hi @sajarin,
Everything looks fine.

**Review by sajarin — APPROVED** [review]

**Comment by sajarin** [comment]
Thanks @Zawar92 for the pr and @cptjacky for the review!

**Pull request merged**
